src/openforms/utils/helpers.py

```diff
@@ -7,12 +9,19 @@ def get_charfield_max_length(
     return model_or_instance._meta.get_field(field_name).max_length
 
 
-def truncate_str_if_needed(original_str: str, new_str: str, max_length: int) -> str:
+def truncate_str_if_needed(
+    original_str: str, new_str: Optional[str], max_length: int
+) -> str:
```

Comment:
I already had trouble reading this function to figure out what exactly it does, and these modifications don't make it easier.

Can't you instead use `django.utils.text.Truncator`?

```py
>>> from django.utils.text import Truncator
>>> Truncator("a string that is too long").chars(20)
'a string that is to…'
```

Probably this existing helper could be converted into that too, but let's leave it untouched for now.

Reply:
I didn't know about it either :open_mouth:. I changed it to use `Truncator` instead

Reply:
it's what powers the template filters like https://docs.djangoproject.com/en/4.2/ref/templates/builtins/#truncatechars :)